pom.xml

```diff
@@ -631,6 +632,26 @@
                             <attachedMavenPlugin>
                                 <originalPluginCoords>io.quarkus:quarkus-maven-plugin:${quarkus.version}</originalPluginCoords>
                                 <targetPluginCoords>${platform.groupId}:quarkus-maven-plugin:${platform.version}</targetPluginCoords>
+                                    <test>
+                                        <!-- Given the Maven plugin is rebuilt as part of the platform project, this is a basic test
+                                             that creates a new Quarkus application project with the 'create' goal targeting the platform version being built
+                                             (which will configure the platform Maven plugin) builds and tests the application with the 'verify` goal
+                                             to run both the unit and the integration tests -->
+                                        <artifact>io.quarkus:quarkus-platform-sample-app-test:${quarkus-platform-bom-generator.version}</artifact>
+                                        <maven-failsafe-plugin>true</maven-failsafe-plugin>
+                                        <systemProperties>
+                                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                                            <quarkus.platform.group-id>${dollarSign}${dollarSign}{platform.groupId}</quarkus.platform.group-id>
+                                            <quarkus.platform.version>${dollarSign}${dollarSign}{platform.version}</quarkus.platform.version>
+                                            <maven.home>${dollarSign}${dollarSign}{maven.home}</maven.home>
+                                            <quarkus.platform.maven-plugin.pom>${dollarSign}${dollarSign}{project.file.absolutePath}</quarkus.platform.maven-plugin.pom>
```

Comment:
I should have added a comment, this option makes sure the POM that is used to resolve the Maven plugin artifact and its dependencies is the flattened version (`.flattened-pom.xml` produced by the flatten-maven-plugin when this module is built) instead of the original `pom.xml`.
So we run the tests using the actual artifacts that will be installed: the platform Maven plugin `quarkus-maven-plugin/target/quarkus-maven-plugin-<version>.jar`, `quarkus-maven-plugin/.flattened-pom.xml`, and the generated `quarkus-bom`.

Reply:
Pushed an update with a comment